byron/ledger/executable-spec/test/Ledger/Delegation/Properties.hs

```diff
@@ -325,88 +331,50 @@ dblockTracesAreClassified = withTests 200 $ property $ do
 
 -- | Extract the delegation certificates in the blocks, in the order they would
 -- have been applied.
+traceDCertsByBlock :: Trace DBLOCK -> [[DCert]]
+traceDCertsByBlock tr = _blockCerts <$> traceSignals OldestFirst tr
+
+-- | Flattended list of DCerts for the given Trace
 traceDCerts :: Trace DBLOCK -> [DCert]
-traceDCerts tr = concat $ _blockCerts <$> traceSignals OldestFirst tr
+traceDCerts = concat . traceDCertsByBlock
 
 relevantCasesAreCovered :: Property
 relevantCasesAreCovered = withTests 400 $ property $ do
   let tl = 1000
   tr <- forAll (trace @DBLOCK tl)
 
-  -- 70% of the traces must contain are as many delegation certificates as
-  -- blocks.
-  cover 50
+  -- 40% of the traces must contain as many delegation certificates as blocks.
+  cover 40
         "there are at least as many delegation certificates as blocks"
         (traceLength tr <= length (traceDCerts tr))
 
-  -- 70% of the traces must contain at most 25% of blocks with empty delegation
-  -- payload.
-  cover 70
+  -- 50% of the traces must contain at most 25% of blocks with empty delegation payload.
+  cover 50
         "at most 25% of the blocks can contain empty delegation payload"
-        (ratio emptyDelegationPayload tr <= 0.25)
+        (0.25 >= emptyDelegationPayloadRatio (traceDCertsByBlock tr))
 
   -- 50% of the traces must contain at least 30% of delegations to this epoch.
   cover 50
-        "at least 30% of the certificates delegate in this epoch"
-        (0.3 <= ratio thisEpochDelegations tr)
+        "at least 30% of all certificates delegate in this epoch"
+        (0.3 <= thisEpochDelegationsRatio (epochDelegationEpoch tr))
 
-  -- 50% of the traces must contain at least 50% of delegations to the next
+  -- 70% of the traces must contain at least 50% of delegations to the next
   -- epoch.
-  cover 50
+  cover 70
         "at least 50% of the certificates delegate in the next epoch"
-        (0.5 <= ratio nextEpochDelegations tr)
+        (0.5 <= nextEpochDelegationsRatio (epochDelegationEpoch tr))
 
-  -- 80% of the traces must contain at least 30% of self-delegations.
-  cover 80
+  -- 30% of the traces must contain at least 30% of self-delegations.
```

Comment:
this is an interesting effect of the re-phrasing of these ratio's we discussed: the ratio is now the number of self-delegs to the total number of certs, whereas before it was the ratio of self-delegs to blocks in the chain

Reply:
Hmmm, but in this case I would expect the ratio to go up (since we would have a smaller denominator) :thinking: 

But anyway, we can leave this for now and give it another look later on.